### ISSUE TITLE
Include Publish for Filesystem

### ISSUE THREAD

**Issue by intellifactory-bb** [intellifactory/websharper#215]
I cannot publish to filesystem (nor Web Deploy Package) - there is no error message only:

```
========== Publish: 0 succeeded, 1 failed, 0 skipped ==========
```

and of course nothing is written.

(Please not: I'm using the 2.5.47-alpha version for .net4 - but I had the same issue with the current nuget sources)

PS: found an error already in the project settings:
![WebsharperPackageError.JPG](https://bitbucket.org/repo/76yqo/images/1744434238-WebsharperPackageError.JPG)

Don't really know if this helps but I guess there might be some missing Interface implementations on some websharper specific components or something (?)

---
- Bitbucket: https://bitbucket.org/intellifactory/websharper/issue/215
- Originally reported by: Carsten König
- Originally created at: 2014-02-10T07:47:25.336

**Comment by cadull** [comment]
For reference, publishing in VS 2015 works, but the script files were missing.

The project file *.fsproj changes mentioned earlier look like:
```
<Target Name="IncludeContent" BeforeTargets="Compile">
  <ItemGroup>
    <Content Include="Content/**/*.*" />
    <Content Include="Scripts/**/*.*" />
  </ItemGroup>
</Target>
```

To include these when publishing, I added the following to the publish settings file *.pubxml:
```
<PropertyGroup>
  <PipelineCollectFilesPhaseDependsOn>
    IncludeContent;
    $(PipelineCollectFilesPhaseDependsOn);
  </PipelineCollectFilesPhaseDependsOn>
</PropertyGroup>
```

**Comment by Tarmil** [comment]
I did a bit of testing and googling around, and it seems that these files are unreliably deployed or not depending on when you build, clean and/or load the project in VS. That's definitely not good.

I think I'm having more luck with replacing that IncludeContent target with the following (based on [this](http://stackoverflow.com/questions/3137880/how-to-get-visual-studio-publish-functionality-to-include-files-from-post-buil)):

```xml
  <PropertyGroup>
    <CopyAllFilesToSingleFolderForPackageDependsOn>
      WebSharperCollectFiles;
      $(CopyAllFilesToSingleFolderForPackageDependsOn);
    </CopyAllFilesToSingleFolderForPackageDependsOn>
  </PropertyGroup>
  <Target Name="WebSharperCollectFiles">
    <ItemGroup>
      <WebSharperContentFiles Include="Content\**\*" />
      <WebSharperScriptsFiles Include="Scripts\**\*" />
      <FilesForPackagingFromProject Include="%(WebSharperContentFiles.Identity)">
        <DestinationRelativePath>Content\%(RecursiveDir)%(Filename)%(Extension)</DestinationRelativePath>
      </FilesForPackagingFromProject>
      <FilesForPackagingFromProject Include="%(WebSharperScriptsFiles.Identity)">
        <DestinationRelativePath>Scripts\%(RecursiveDir)%(Filename)%(Extension)</DestinationRelativePath>
      </FilesForPackagingFromProject>
    </ItemGroup>
  </Target>
```

Can you confirm that this works for you as well? If so, then we should use this in the project templates; or maybe even include it in `WebSharper.targets` instead.

**Comment by cadull** [comment]
Your suggestion worked for me when publishing to the file system, but not with a web deploy.

**Comment by Tarmil** [comment]
Indeed, and there's a link to [this documentation](https://docs.microsoft.com/en-us/aspnet/web-forms/overview/deployment/visual-studio-web-deployment/deploying-extra-files) which uses both.

**Comment by cadull** [comment]
Including both is working for publish to file system and web deploy.

    <PropertyGroup>
      <CopyAllFilesToSingleFolderForPackageDependsOn>
        WebSharperCollectFiles;
        $(CopyAllFilesToSingleFolderForPackageDependsOn);
      </CopyAllFilesToSingleFolderForPackageDependsOn>
      <CopyAllFilesToSingleFolderForMsdeployDependsOn>
        WebSharperCollectFiles;
        $(CopyAllFilesToSingleFolderForMsdeployDependsOn);
      </CopyAllFilesToSingleFolderForMsdeployDependsOn>
    </PropertyGroup>
    <Target Name="WebSharperCollectFiles">
      <ItemGroup>
        <WebSharperContentFiles Include="Content\**\*" />
        <WebSharperScriptsFiles Include="Scripts\**\*" />
        <FilesForPackagingFromProject Include="%(WebSharperContentFiles.Identity)">
          <DestinationRelativePath>Content\%(RecursiveDir)%(Filename)%(Extension)</DestinationRelativePath>
        </FilesForPackagingFromProject>
        <FilesForPackagingFromProject Include="%(WebSharperScriptsFiles.Identity)">
          <DestinationRelativePath>Scripts\%(RecursiveDir)%(Filename)%(Extension)</DestinationRelativePath>
        </FilesForPackagingFromProject>
      </ItemGroup>
    </Target>
